src/lib.rs

```diff
@@ -14,6 +14,16 @@
 //! This feature currently enables the `std` feature, meaning that it is not
 //! compatible with `no_std` environments.
 //!
+//! ### `msrv` (default)
+//!
+//! Enables Rust language features that requires a higher MSRV. Enabling this
+//! feature on older compilers will NOT result in a compilation error, the newer
+//! language features will simply not be used.
+//!
+//! Right now, this feature only enables the use of
+//! `diagnostic::on_unimplemented` (Rust 1.78 or later) to provide better error
+//! messages.
```

Comment:
```suggestion
//! When compiling with Rust v1.78 or later, this feature enables better error messages for invalid methods on structs and enums.
```
Lets leave implementation details out of public documentation. Feel free to re-word, you get the gist.